flang/lib/Lower/Bridge.cpp

```diff
@@ -336,7 +340,7 @@ class FirConverter : public Fortran::lower::AbstractConverter {
 
   bool createHostAssociateVarClone(
       const Fortran::semantics::Symbol &sym) override final {
-    auto loc = genLocation(sym.name());
+    mlir::Location loc = genLocation(sym.name());
```

Comment:
As @schweitzpgi mentioned MLIR is using `auto` when they retrieve a `mlir::Location` -> `getLoc()`. Since it's not `operation->getLoc()` here using `mlir::Location` can be fine too.